README.md

```diff
@@ -54,9 +54,9 @@ For more details on the yotta/mbed OS edition of mbed TLS, including example pro
 
 ### Make
 
-We intentionally only use the minimum of `Make` functionality, as a lot of `Make` features are not supported on all different implementations of Make or on different platforms. As such, the Makefiles sometimes require some manual changes or export statements in order to work for your platform.
+We require GNU Make. To build the library and the sample programs, GNU Make and a C compiler are sufficient. Some of the more advanced build targets require some Unix/Linux tools.
```

Comment:
So we only test with GNU Make and guarantee it out of the box. I think we also need a statement to say we aim to be cross-platform, and it should build on Linux, BSD and Windows.

I think we can also still retain the aspiration to keep the Makefiles simple, to allow them to be more easily ported.

Therefore I'd put back the line: 

> We intentionally only use the minimum of `Make` functionality, as a lot of `Make` features are not supported on all different implementations of Make or on different platforms. As such, the Makefiles sometimes require some manual changes or export statements in order to work for your platform.

Reply:
Hi @gilles-peskine-arm. Re-reading this I don't think we're discussing the same point. Looking at a different PR, #568, I think @mpg made the same, if not very similar, point much more eloquently in a review comment on the makefiles.

Deliberately using only your original words and @mpg's to try and capture the original spirit of the paragraph I was trying to retain, I propose the following:

```
### GNU Make

We require GNU Make. To build the library and the sample programs, GNU Make and a C compiler are sufficient. Some of the more advanced build targets require some Unix/Linux tools.

We intentionally only use a minimum of `Make` functionality, to keep the makefiles as simple and independent of different toolchains as possible, to allow users to more easily move between different platforms. Users who need more features are recommended to use CMake.
```

What do you think?

Reply:
@sbutcher-arm Thanks Simon! You make a good point of keeping a note about using functionality. I've almost used your word, but I tweaked them to say “functionality in the makefiles” rather than “make functionality”, because it's about the overall functionality (make, shell, utilities, toolchains) and not make specifically.